### PR TITLE
Faucet Polling in E2E Tests

### PR DESCRIPTION
**Pull request by usmanmani1122**
## Description
Updated test to poll for transaction completion
Successful run on Emerynet:
https://github.com/Agoric/dapp-psm/actions/runs/10792624611/job/29932687814

### CODE REVIEW

File: tests/e2e/support.js [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
+          const { transactionStatus } = resp.body;
+          if (transactionStatus === TRANSACTION_STATUS.NOT_FOUND)
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            return cy.wait(2000).then(() => getStatus(txHash));
+          else return cy.wrap(transactionStatus);
+        });
+
     cy.request({
-      method: 'POST',
-      url: networkPhrases.faucetUrl,
```
Comment (frazarshad) [inline tests/e2e/support.js:11]: nit: i think its safe to remove `faucetUrl` from `networkPhrases` now

File: tests/e2e/support.js [hunk 2]
```diff
@@ -1,24 +1,45 @@
...
+          method: 'GET',
+          url: `https://${Cypress.env(
+            'AGORIC_NET'
+          )}.faucet.agoric.net/api/transaction-status/${txHash}`,
+        })
+        .then(resp => {
+          const { transactionStatus } = resp.body;
+          if (transactionStatus === TRANSACTION_STATUS.NOT_FOUND)
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            return cy.wait(2000).then(() => getStatus(txHash));
```
Comment (Muneeb147) [inline tests/e2e/support.js:25]: Do we have in general timeout limit on our tests? Thinking of a case when we are getting consistent `NOT_FOUND` for some unexpected reason.

**Review by Muneeb147 — APPROVED** [review]
LGTM

File: tests/e2e/support.js (hunk 2 above)
Comment (usmanmani1122) [reply tests/e2e/support.js:25]: Yes there is a maximum timeout limit of 30 seconds

**Pull request merged**